### DISCUSSION TITLE
Can't get dynamic paths to work with ISR

### DISCUSSION THREAD

**Discussion by kmcaloon** [vercel/next.js#16731] [Help] [answered]
I'm not sure if what I am trying to do is possible or if I am completely understanding dynamic paths. Basically I would like to set up my site so that is completely dynamic with ISG that regenerates pages say every minute. I am hoping to not have to prerender dynamic paths during build, but rather just have next+vercel generate and cache them on demand. Here is an example of a dynamic page I have set up as `pages/drone-training/[course].js`

```
export async function getStaticPaths() {
  return {
    paths: [],
    fallback: true,
  }
}
export async function getStaticProps( { params } ) {

  const { course } = params;
  const sanityClient = require( 'lib/sanity' ).client;
  const queryParams = {
    course,
  }

  const response = await sanityClient.fetch( pageQuery, queryParams );

  return {
    props: {
      ...response,
    },
    revalidate: 60,
  }
}
```

Everything is working fine locally, but when building it out next generates this page as an `SSG` page instead of `ISR`. It looks like `revalidate` is being ignored, and my hunch is that it has something to do with my `getStaticPaths` set up. Hopefully what I am trying to pull off is possible.

Thanks

**Comment by alanqchen** [comment, marked as answer]
BTW when building, ISR pages show the same icon (filled-in circle) as SSG pages. To confirm that ISR is (or isn't) working, I would try doing the following:

Add a updatedAt field to your returned props in `getStaticProps`:
```
props: {
     ...response,
     updatedAt: Date.now()
}
```
Then in your page component add the time:

```
export default function SomePage({ updatedAt }) {
    const timeString = new Date(updatedAt).toLocaleTimeString();

     return (
         <p>{timeString}</p>
     )
}
```

([Bascially the second example in this article](https://arunoda.me/blog/what-is-nextjs-issg)).

If the time updates on every refresh on the production build (or never updates), then we can confirm that ISR is not working.
But if it updates on the second reload after the revalidate time has passed, then ISR is working

**Reply by kmcaloon** [reply]
Thanks @alejandronanez so I guess I'm confused about this

```
17:12:32.209 | ├ ● /drone-training/[course]                                   291 kB          376 kB
```

and this

```
17:12:32.209 | λ  (Lambda)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
-- | --
17:12:32.209 | ○  (Static)  automatically rendered as static HTML (uses no initial props)
17:12:32.209 | ●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)
17:12:32.209 | (ISR)     incremental static regeneration (uses revalidate in getStaticProps)
```

It makes it seem as if only pages without a dot are exported with ISR.

A further development that is driving me crazy, is that I am not seeing refreshed content in my production environment even when I redeploy everything multiple times. I have changed the content in my CMS, and added those props you had mentioned, the builds are succeeding and deploy on Vercel, yet when I pull up the page it shows the old content without the new `updatedAt` prop.  In other words, even after deploy it seems to just be showing a stale cached page: https://dartdrones-website.vercel.app/drone-training/flying-class

Not really sure where to go from here?

**Comment by kmcaloon** [comment]
So there were a couple of things going on here. First, @alanqchen you were right, I was confused by the build notes. They make it seem as if anything with a dot will not be a candidate for ISR.

The setup above actually worked with ISR and completely dynamic paths. The other issue I was running into was I had old Gatsby cache in my `public` directory as I am switching between branches and forgot to clear it out. Long story short this is why I wasn't seeing a refreshed page after the revalidate interval.

ISR is pretty awesome!

**Comment by spencersmb** [comment]
I'm having a similar issue - I just have an index page with getStaticProps and I'm trying to get it to revalidate in production and I get nothing. When running npm run build and then npm run start - it works locally just fine. I'm wondering what needs to be done in production to enable it? 

Any help is much appreciated!
Thanks,
Spencer

**Reply by clementpeleman** [reply]
Same problem Spencer!

**Reply by alexandrosk** [reply]
Any luck you guys? Im having the same issue.

**Reply by clementpeleman** [reply]
Yea, I fixed it! I removed the ApolloClient I was using for the data fetching from the CMS. And also added “next build” in the vercel build command
![6F727EE1-22B6-4897-B3FD-A6D07C6C5364](https://user-images.githubusercontent.com/28512676/148116384-7971e9d1-67ed-4456-a656-7e2f642aeb8d.png)

**Reply by clementpeleman** [reply]
Apollo was causing troubles due to the caching. I think the main thing is the vercel command tho as it is possible to disable caching in apollo

**Reply by alexandrosk** [reply]
Hehe crazy, I just found it and then I read your comments, it was only making sense to be apollo! 
Thanks a lot.

**Reply by clementpeleman** [reply]
Glad to hear :)
